pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -275,11 +275,24 @@ func (r *RestartReconciler) reconcileNodes(ctx context.Context) (ctrl.Result, er
 
 // restartPods restart the down pods using an admin command
 func (r *RestartReconciler) restartPods(ctx context.Context, pods []*PodFact) (ctrl.Result, error) {
-	// Reduce the pod list according to the cluster node state
-	downPods, res, err := r.removePodsWithClusterUpState(ctx, pods)
+	// we want to fetch the state of down pods and primary pods. The letter will
+	// allow us to check cluster quorum and abort restart_node if needed
+	podsToFetch := combinePodsWithoutDuplicates(r.PFacts.findPrimaryPods(), pods)
```

Comment:
This logic in here is getting complicated. We may be able to simplify this by just getting the cluster node status for all pods. Basically don't do any filtering in FetchNodeState. We are given a map, which we can do fast lookups on, so there isn't a large gain to having the map filtered.

Reply:
I was just worried about having a map too big. But FetchNodeState() already returns the entire cluster state so there is no benefits. I will remove that.

Now I am wondering why we have a `HostsNeeded` map in fetchnode config.

Reply:
It was probably an optimization that didn't make sense to do, which was probably my doing (:smile:).